tls/ttls.c

```diff
@@ -1141,12 +1126,23 @@ ttls_parse_record_hdr(TlsCtx *tls, unsigned char *buf, size_t len,
 	}
 
 	io->msgtype = io->hdr[0];
-	tls->major = io->hdr[1];
-	tls->minor = io->hdr[2];
+	/*
+	 * The only valid major version accordingly to RFC 5246 (TLS 1.2) and
+	 * RFC 8446 (TLS 1.3) is 0x03, so treat any different values as an
+	 * attempt to confuse us and block such records.
+	 *
+	 * However RFC 8446 5.1 and D.2 suggest to tolerate minor versions
+	 * 0x01 for compatibility reasons. In fact OpenSSL may send 0x0301
+	 * with ClientHello. The higher minor version is still 0x03.
```

Comment:
```suggestion
	 * in client records. The higher minor version is still 0x03.
```

Seems like the comment is bit misleading. This specific checks are imply protocol version in the record header, while ClientHello has an additional field to show the real protocol version. But yes, OpenSSL tends to send `0x301` protocol version in all the client records.